lib/galaxy/managers/libraries.py

```diff
@@ -112,7 +112,11 @@ def list(self, trans, deleted=False):
         """
         is_admin = trans.user_is_admin()
         query = trans.sa_session.query(trans.app.model.Library)
-
+        library_access_action = trans.app.security_agent.permitted_actions.LIBRARY_ACCESS.action
+        restricted_library_ids = [lp.library_id for lp in (
```

Comment:
If these can grow really big it might be beneficial to make it a set
```python
        restricted_library_ids = {lp.library_id for lp in (
            trans.sa_session.query(trans.model.LibraryPermissions).filter(
                trans.model.LibraryPermissions.table.c.action == library_access_action
            ).distinct())}
```

Reply:
Could you add the restricted_library_ids list/set to the docstring ?

Reply:
@mvdbeek There is `.distinct()` in the query that I hope has similar effect, but set does not make this worse I think. Will switch.